### ISSUE TITLE
"custom from" can lead to invalid "from" adresses

### ISSUE THREAD

**Issue by codegain** [roundcube/roundcubemail#8106]
Hi,

I'm using Roundcube 1.4.11 on PHP 7.4 and stumbled upon an error message from our smtp server:

```
Failed to set sender '"name@domain'. <"name@domain>: unmatched doublequote in local part (Code: 501)
```

I did some debugging and found out that using `name@domain .com` as display name configured in roundcube triggers it. This is because in `rcmail_sendmail.php:140` there is some code to find the `from` adress:

```
        // Get sender name and address from identity...
        if (is_numeric($from)) {
            if (is_array($identity_arr = $this->get_identity($from))) {
                if ($identity_arr['mailto']) {
                    $from = $identity_arr['mailto'];
                }
                if ($identity_arr['string']) {
                    $from_string = $identity_arr['string'];
                }
            }
            else {
                $from = null;
            }
        }
        // ... if there is no identity record, this might be a custom from
        else if (($from_string = $this->email_input_format($from))
            && preg_match('/(\S+@\S+)/', $from_string, $m)
        ) {
            $from = trim($m[1], '<>');
        }
        // ... otherwise it's empty or invalid
        else {
            $from = null;
        }
```
(In my case, the `from` value is not read from the identity because there is no identity record)

In the part `// ... if there is no identity record, this might be a custom from` `$from` is set from the first match of `/(\S+@\S+)/` without any further validity checks.
This causes `$from` to become `"name@domain` as shown with this code:

```
$from_string = '"name@domain .com" <name@domain.com>';
 
preg_match('/(\S+@\S+)/', $from_string, $m);
 
var_dump($m);
```
which outputs:
```
array(2) {
  [0]=>
  string(14) ""name@domain"
  [1]=>
  string(14) ""name@domain"
}
```

**Comment by alecpl** [comment]
Fixed.

**Comment by codegain** [comment]
@alecpl Hi, I tested the fix in the latest 1.4.12 release and it turns out that this is no longer working for umlaut-domains like "myuser@mydomäin.com". In the previous code there was 

```
$this->email_input_format($from)
```

which converted the email from "myuser@mydomäin.com" to "myuser@xn--mydomin-9wa.com". 

I could fix it on my end with the following code:

```
// ... if there is no identity record, this might be a custom from
$from_addresses = rcube_mime::decode_address_list($this->email_input_format($from));
```
